packages/react-native-web/src/exports/Image/ImageUriCache.js

```diff
@@ -7,57 +7,80 @@
  * @flow
  */
 
-const dataUriPattern = /^data:/;
+import ImageLoader from '../../modules/ImageLoader';
+
+type ImageSource =
+  | string
+  | number
+  | {
+      method: ?string,
+      uri: ?string,
+      headers: ?Object,
+      body: ?string
+    };
 
 export default class ImageUriCache {
   static _maximumEntries: number = 256;
   static _entries = {};
 
-  static has(uri: string) {
+  static createCacheId(source: ImageSource) {
+    return JSON.stringify(ImageLoader.resolveSource(source));
```

Comment:
I'm a little concerned about using `JSON.stringify` here. We could use a Map instead

Reply:
Yeah, I don't like it either but I could not find a smart way. The advantage of this approach is that you will always get the cached content for a combination of method, uri, headers and body. If we use a Map instead and use the source as a key then we somehow first need to get the reference to the source which leads to the same issue again. Or did you have something else in mind?

Reply:
A `Map` can use objects as keys

Reply:
Yes, it allows object as keys but the reference is stored as a key and not the content of the object:
```ts
const map = new Map()
const key = { a: 1 }
map.set(key, true)

map.get(key) !== map.get({ a: 1 }) // true
```

Reply:
Yeah that's a problem given how image sources are defined. I guess we're stuck with stringifying for now